packages/astro/src/i18n/index.ts

```diff
@@ -19,12 +19,13 @@ export function requestHasLocale(locales: Locales) {
 export function pathHasLocale(path: string, locales: Locales): boolean {
 	const segments = path.split('/');
 	for (const segment of segments) {
+		const normalizedSegment = normalizeThePath(segment);
```

Comment:
This function is only used here, so there's no need to have a function. We can add a `.map` after the `split` at line 20 and have the mapping logic there.

Also, since this happens for a specific case e.g. `fr-fr.html`, we could leave a comment that explains the `.html` check